### PR TITLE
feat: Added meter chart derived from donut chart

### PR DESCRIPTION
**Pull request by maulik-thaker1**
## Summary

- Added meter chart derived from donut chart and adjust plugin calls and chart type accordingly

## ADO Story

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1670704

## Notes

- Created one plugin to draw and animate Needle named `meterGaugeNeedle.js`
- ChartJS doesn't have `type = 'meter'` so we have to make adjustment inside `chart.ts` file while selecting type prop. and insert only Meter chart related plugin when type='meter' is called otherwise doughnut plugin(s) are also called and mixed up with Meter plugins 

## To Do

- Required by any from UX side

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [x] Added/updated a basic unit test, at least, for each component.
- [x] Added/updated the Figma link for the Story's Design tab.
- [x] Added any new component exports to the src/index.ts file

### CODE REVIEW

File: src/stories/Meter.stories.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+      cutout: '80%',
+      // Required needleValue to show needle
+      needleValue: 40,
+      //   cutoutPercentage: 50,
+    },
+  ],
+  options: {
+    plugins: {
+      aspectRatio: 1.7,
```
Comment (brian-patrick-3) [inline src/stories/Meter.stories.js:37]: This should be under `options`, not `plugins`, it doesn't work here. You should also move it to `/config/chartTypes/meter.js`

File: src/stories/Meter.stories.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+const args = {
+  chartTitle: 'Meter Chart',
+  description: 'Chart description.',
+  labels: ['LOW', 'MED', 'HIGH'],
+  datasets: [
+    {
+      data: [30, 30, 30],
+      borderWidth: 2,
+
+      cutout: '80%',
```
Comment (brian-patrick-3) [inline src/stories/Meter.stories.js:29]: This should move to `/config/chartTypes/meter.js` also.

File: src/stories/Meter.stories.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+};
+
+const args = {
+  chartTitle: 'Meter Chart',
+  description: 'Chart description.',
+  labels: ['LOW', 'MED', 'HIGH'],
+  datasets: [
+    {
+      data: [30, 30, 30],
+      borderWidth: 2,
```
Comment (brian-patrick-3) [inline src/stories/Meter.stories.js:27]: Move to `/config/chartTypes/meter.js`

File: src/stories/Meter.stories.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      needleValue: 40,
+      //   cutoutPercentage: 50,
+    },
+  ],
+  options: {
+    plugins: {
+      aspectRatio: 1.7,
+      legend: {
+        display: false,
+      },
```
Comment (brian-patrick-3) [inline src/stories/Meter.stories.js:40]: Move to `/config/chartTypes/meter.js`

File: src/stories/Meter.stories.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  //     animation: {
+  //       animateRotate: true,
+  //       animateScale: false,
+  //     },
+  //   },
+  hideDescription: false,
+  hideCaptions: false,
+  hideHeader: false,
+  hideControls: false,
+  colorPalette: 'rag03',
```
Comment (brian-patrick-3) [inline src/stories/Meter.stories.js:87]: Set default color palette in `/config/chartTypes/meter.js`

File: src/common/config/chartTypes/meter.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+          return context.chart.data.labels[context.dataIndex];
+        },
+      },
+    },
+  };
+};
+
+export const datasetOptions = (ctx, index) => {
+  return {
+    backgroundColor: colorPalettes[ctx.options.colorPalette || 'categorical'],
```
Comment (brian-patrick-3) [inline src/common/config/chartTypes/meter.js:38]: Default color palette can be set here.

File: src/common/config/chartTypes/meter.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    radius: '80%',
+    circumference: 180,
+    rotation: 270,
+    plugins: {
+      datalabels: {
+        color: LabelColor,
+        font: function (context) {
+          var w = context.chart.width;
+          return {
+            size: w < 512 ? 10 : 12,
```
Comment (brian-patrick-3) [inline src/common/config/chartTypes/meter.js:21]: Is it necessary to have variable font size here?

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
Seems pretty good overall, just need to move the default settings to the config file, and make sure design/Robert approves.

File: src/stories/Meter.stories.js (hunk 2 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js:29]: But `borderWidth` and `cutout` is modifiable by user so I put here.

File: src/stories/Meter.stories.js (hunk 3 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js:27]: `data: [30,30,30] `will be part of stories correct ?

File: src/common/config/chartTypes/meter.js (hunk 7 above)
Comment (maulik-thaker1) [reply src/common/config/chartTypes/meter.js:21]: I added this for responsive screen as we are displaying legend on meter area hence I made font 12px for large screen and 10px for small / mobile screen

File: src/stories/Meter.stories.js (hunk 2 above)
Comment (brian-patrick-3) [reply src/stories/Meter.stories.js:29]: If those are defaults they need to go in the config file. Technically all options are user modifiable, so no need to use them in a story unless there's a specific variant you want to show .

File: src/stories/Meter.stories.js (hunk 3 above)
Comment (brian-patrick-3) [reply src/stories/Meter.stories.js:27]: Yeah data is always user supplied

File: src/stories/Meter.stories.js (hunk 2 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js:29]: oh can they modify the config file ?

File: src/stories/Meter.stories.js (hunk 3 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js:27]: ok then it's here only inside stories.js

File: src/stories/Meter.stories.js (hunk 2 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js:29]: done

File: src/stories/Meter.stories.js (hunk 1 above)
Comment (maulik-thaker1) [reply src/stories/Meter.stories.js]: done

**Comment by maulik-thaker1** [comment]
Waiting for @kaanhsu  / UX's feedback.

File: src/stories/Meter.stories.js (hunk 2 above)
Comment (brian-patrick-3) [reply src/stories/Meter.stories.js]: > oh can they modify the config file ?

No. They can override any default option though.

Comment (maulik-thaker1) [reply src/stories/Meter.stories.js]: okay

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by maulik-thaker1** [comment]
> Waiting for @kaanhsu / UX's feedback.

Approved by @kaanhsu hence merged this.

**Pull request merged**